components/micro-gateway-jwt-generator/src/main/java/org/wso2/micro/gateway/jwt/generator/AbstractMGWJWTGenerator.java

```diff
@@ -298,29 +303,23 @@ public String addCertToHeader() throws Exception {
         String base64UrlEncodedThumbPrint;
         base64UrlEncodedThumbPrint = java.util.Base64.getUrlEncoder()
                 .encodeToString(publicCertThumbprint.getBytes("UTF-8"));
-        StringBuilder jwtHeader = new StringBuilder();
         //Sample header
         //{"typ":"JWT", "alg":"SHA256withRSA", "x5t":"a_jhNus21KVuoFx65LmkW2O_l10"}
         //{"typ":"JWT", "alg":"[2]", "x5t":"[1]"}
-        jwtHeader.append("{\"typ\":\"JWT\",");
-        jwtHeader.append("\"alg\":\"");
-        jwtHeader.append("RS256");
-        jwtHeader.append("\",");
-
-        jwtHeader.append("\"x5t\":\"");
-        jwtHeader.append(base64UrlEncodedThumbPrint);
-        jwtHeader.append('\"');
-
-        jwtHeader.append('}');
-
+        Map<String, String> jsonMap = new HashMap<>();
+        jsonMap.put("typ", "JWT");
```

Comment:
shall we move these also to constant file?

Reply:
Done with https://github.com/wso2/product-microgateway/pull/1613/commits/4a594ae1d7dfcf91fe90c9c81f90e9d83c03a972